lab_1/main.py

```diff
@@ -6,12 +6,20 @@
 
 def tokenize(text: str) -> list or None:
     """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation and other symbols from words
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+        Splits a text into tokens, converts the tokens into lowercase,
+        removes punctuation and other symbols from words
+        :param text: a text
+        :return: a list of lower-cased tokens without punctuation
+        """
+    if isinstance(text, str):
```

Comment:
this check can be rewritten to "if not" statement to simplify functions and avoid unnecessary nesting

Reply:
i've rewritten nearly all parts with 'if' conditions, thank you for the comments!